Doc/library/shutil.rst

```diff
@@ -263,8 +262,41 @@ Directory and files operations
 
    If *copy_function* is given, it must be a callable that will be used to copy
    each file. It will be called with the source path and the destination path
-   as arguments. By default, :func:`~shutil.copy2` is used, but any function
-   that supports the same signature (like :func:`~shutil.copy`) can be used.
+   as arguments. By default, :func:`~shutil.copy2` is used, but any function
+   that supports the same signature (like :func:`~shutil.copy`) can be used.
+
+   If *dirs_exist_ok* is true, existing directories along the path to *dst*
+   will be overwritten.  If *dirs_exist_ok* is false (the default) and *dst*
+   already exists, a :exc:`FileExistsError` is raised. Internally, this
+   parameter is passed directly to :func:`os.makedirs`.
+
+   .. note::
+
+      *dirs_exist_ok* specifically refers to whether *already existing*
+      directories should be overwritten. However, the copytree function will
+      always create paths if they do not exist. For example, consider the
+      following file tree::
+
+         |-- source
+         |  |-- foo.txt
+         |  |-- bar
+         |     |-- baz.txt
+         |     |-- buzz.py
+         |
+         |-- neighbor
+            |-- cat.png
+
+      Calling :func:`shutil.copytree` like this::
+
+         shutil.copytree('source', 'these/all/do/not/exist/yet')
+
+      Will create all those intermediary folders, even if *dirs_exist_ok* is
+      false.
+
+      However, the following will cause a :exc:`FileExistsError`, because it
```

Comment:
Could this mention how the behavior depends on whether `dir_exist_ok` is passed? As written the example doesn't explicitly explain where it makes a difference.

Reply:
Also, this example feels a bit excessive; we generally aim for concision in the docs. I see that lots of people have already contributed to this discussion though, so if they're in general agreement I'm OK with adding it.

Reply:
Yeah, I'm not crazy about the lack of brevity in it either, anymore. This was one of my first contributions and I was feeling myself a bit too much –– 

I will fix this up sometime soon when I get a chance. Thank you for the ping.